ws/adapter_test.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Mainflux
+// SPDX-Licence-Identifier: Apache-2.0
+
+package ws_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/mainflux/mainflux"
+	httpmock "github.com/mainflux/mainflux/http/mocks"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/ws"
+	"github.com/mainflux/mainflux/ws/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	chanID   = "1"
+	id       = "1"
+	thingKey = "thing_key"
+	subTopic = "subtopic"
+	protocol = "ws"
+)
+
+var msg = messaging.Message{
+	Channel:   chanID,
+	Publisher: id,
+	Subtopic:  "",
+	Protocol:  protocol,
+	Payload:   []byte(`[{"n":"current","t":-5,"v":1.2}]`),
+}
+
+func newService(cc mainflux.ThingsServiceClient) (ws.Service, mocks.MockPubSub) {
+	pubsub := mocks.NewPubSub()
+	return ws.New(cc, pubsub), pubsub
+}
+
+func NewThingsClient() mainflux.ThingsServiceClient {
```

Comment:
No need for this wrapper, call `httpmock.NewThingsClient(map[string]string{thingKey: chanID})` directly.